python-setup/auto_install_packages.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+
+import sys
+import os
+import subprocess
+from tempfile import mkdtemp
+
+import extractor_version
+
+
+def _check_call(command):
+    print('+ {}'.format(' '.join(command)), flush=True)
+    subprocess.check_call(command, stdin=subprocess.DEVNULL)
+
+
+def _check_output(command):
+    print('+ {}'.format(' '.join(command)), flush=True)
+    out = subprocess.check_output(command, stdin=subprocess.DEVNULL)
+    print(out, flush=True)
+    sys.stderr.flush()
+    return out
+
+
+def install_packages_with_poetry():
+    try:
+        _check_call(['poetry', 'install', '--no-root'])
+    except subprocess.CalledProcessError:
+        sys.exit('package installation with poetry failed, see error above')
+
+    # poetry is super annoying with `poetry run`, since it will put lots of output on
+    # STDOUT if the current global python interpreter is not matching the one in the
+    # virtualenv for the package, which was the case for using poetry for Python 2 when
+    # default system interpreter was Python 3 :/
+
+    poetry_out = _check_output(['poetry', 'run', 'which', 'python'])
+    python_executable_path = poetry_out.decode('utf-8').splitlines()[-1]
+
+    return python_executable_path
+
+
+def install_packages_with_pipenv():
+    try:
+        _check_call(['pipenv', 'install', '--keep-outdated', '--ignore-pipfile'])
+    except subprocess.CalledProcessError:
+        sys.exit('package installation with pipenv failed, see error above')
+
+    pipenv_out = _check_output(['pipenv', 'run', 'which', 'python'])
+    python_executable_path = pipenv_out.decode('utf-8').splitlines()[-1]
+
+    return python_executable_path
+
+
+def _create_venv(version: int):
+    # create temporary directory ... that just lives "forever"
+    venv_path = mkdtemp(prefix='codeql-action-python-autoinstall-')
+
+    # virtualenv is a bit nicer for setting up virtual environment, since it will provide
+    # up-to-date versions of pip/setuptools/wheel which basic `python3 -m venv venv` won't
+
+    if version == 2:
+        _check_call(['python2', '-m', 'virtualenv', venv_path])
+    elif version == 3:
+        _check_call(['python3', '-m', 'virtualenv', venv_path])
```

Comment:
Can `get_extractor_version` return anything else apart from `2` or `3`? If it can, then we do not install the virtual env, right? Is this expected? If it cannot then I would change this into:
```suggestion
    else:
        assert version == 3
        _check_call(['python3', '-m', 'virtualenv', venv_path])
```

Reply:
it will only return `2` or `3`.